tests/Fixtures/TestBundle/Entity/Issue6212/Nest.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Tests\Fixtures\TestBundle\Entity\Issue6212;
+
+use ApiPlatform\Metadata\ApiResource;
+use ApiPlatform\Tests\Fixtures\TestBundle\ApiResource\Issue6212\Bird;
+use ApiPlatform\Tests\Fixtures\TestBundle\ApiResource\Issue6212\Robin;
+use ApiPlatform\Tests\Fixtures\TestBundle\ApiResource\Issue6212\Wren;
+use Doctrine\ORM\Mapping as ORM;
+
+#[ApiResource]
+#[ORM\Entity]
+class Nest
+{
+    #[ORM\Id]
+    #[ORM\GeneratedValue]
+    #[ORM\Column]
+    private ?int $id = null;
+
+    #[ORM\Column(type: 'json')]
+    private ?Bird $owner;
+
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    public function getOwner(): ?Bird
+    {
+        return $this->owner;
+    }
+
+    public function setOwner(Wren|Robin|null $owner): static
```

Comment:
```suggestion
    public function setOwner(?Bird $owner): static
```

Reply:
Agreed that in this tiny model that looks wrong, but it is just meant to illustrate the bug. In this case, the concrete classes implement the `Bird` interface, but the `Robin` classes have an extra property. Normally I'd use a second interface but didn't want to overload the example/test :innocent: 

As I understand the flow, the `PropertyAccessor` uses the types on the setter (as a first option), and the additional (non-`null`) properties are currently not properly handled in the schema factory. So it is only the setter in this case using that signature.